### PR TITLE
Added method for sorted print

### PR DESCRIPTION
**Pull request by mattinbits**
This method prints in the same format
as PrettyPrinter but sorts the keys
of each object lexicographically.

The impetus for this change was this
question:
http://stackoverflow.com/questions/31418626/sort-fields-in-rendered-json

In general it is useful to be able to have more deterministic
control over the ordering of output, if diff tools
are to be used.

As per the README, I'm happy to license the work to the spray-json project under the project’s open source license. Wasn't sure what to put as a licence header in each file. If you let me know, I'll update the PR accordingly.

### CODE REVIEW

**Comment by ktoso** [comment]
Resolves https://github.com/spray/spray-json/issues/224

File: src/main/scala/spray/json/JsValue.scala [hunk 1]
```diff
@@ -28,6 +28,7 @@ sealed abstract class JsValue {
   def toString(printer: (JsValue => String)) = printer(this)
   def compactPrint = CompactPrinter(this)
   def prettyPrint = PrettyPrinter(this)
+  def sortedPrint = SortedPrinter(this)
```
Comment (ktoso) [inline src/main/scala/spray/json/JsValue.scala]: this should be fine, no one is supposed to extend this class other than spray-json itself

File: src/main/scala/spray/json/SortedPrinter.scala [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package spray.json
```
Comment (ktoso) [inline src/main/scala/spray/json/SortedPrinter.scala]: I will add license headers

File: src/main/scala/spray/json/SortedPrinter.scala [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+trait SortedPrinter extends PrettyPrinter {
+
+  override protected def organiseMembers(members: Map[String, JsValue]): Seq[(String, JsValue)] =
```
Comment (ktoso) [inline src/main/scala/spray/json/SortedPrinter.scala]: ok

File: src/main/scala/spray/json/SortedPrinter.scala [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+    members.toSeq.sortBy(_._1)
+}
```
Comment (ktoso) [inline src/main/scala/spray/json/SortedPrinter.scala]: ok should be fine

**Comment by tonyamelnychuk** [comment]
Hi guys, should CompactPrinter also has such ordering?

**Comment by mattinbits** [comment]
Hi @tonyamelnychuk it was not a goal of this PR to do that, but I can imagine this could be useful functionality in other use cases. Given this is an old, closed PR, the best course of action may be to create a new issue or PR for the requirement.

**Pull request merged**